wgpu-core/src/present.rs

```diff
@@ -103,17 +73,17 @@ pub enum CreateSwapChainError {
 
 #[repr(C)]
 #[derive(Debug)]
-pub struct SwapChainOutput {
+pub struct SurfaceOutput {
     pub status: Status,
     pub view_id: Option<TextureViewId>,
 }
 
 impl<G: GlobalIdentityHandlerFactory> Global<G> {
-    pub fn swap_chain_get_current_texture_view<A: HalApi>(
+    pub fn surface_get_current_texture_view<A: HalApi>(
```

Comment:
Shouldn't this return a texture, not a texture view, as we discussed (per https://www.w3.org/TR/webgpu/#dom-gpucanvascontext-getcurrenttexture)?  Then you can get rid of the `TextureViewSource::Surface` variant (and `TextureViewSource` more generally)--and consequently also get rid of the rogue texture view destruction stuff.  I think we also determined there's no actual reason the texture needs to know it came from a swapchain, so this doesn't have to get replaced with something on `Texture`.

Reply:
Yes, it's just a different piece of work, fairly big one, so I didn't want to mix them in the same change. I'm working on it.